EIPS/eip-7002.md

```diff
@@ -0,0 +1,380 @@
+
+As currently specified, only the active key can initiate a validator exit. This means that in any non-standard custody relationships (i.e. active key is separate entity from withdrawal credentials), that the ultimate owner of the funds -- the possessor of the withdrawal credentials -- cannot independently choose to exit and begin the withdrawal process. This leads to either trust issues (e.g. ETH can be "held hostage" by the active key owner) or insufficient work-arounds such as pre-signed exits. Additionally, in the event that active keys are lost, a user should still be able to recover their funds by using their cold withdrawal credentials.
+
+To ensure that the withdrawal credentials (owned by both EOAs and smart contracts) can trustlessly control the destiny of the staked ETH, this specification enables exits triggerable by 0x01 withdrawal credentials.
+
+Note, 0x00 withdrawal credentials can be changed into 0x01 withdrawal credentials with a one-time signed message. Thus any functionality enabled for 0x01 credentials is defacto enabled for 0x00 credentials.
+
+## Specification
+
+### Constants
+
+| Name | Value | Comment |
+| - | - | - |
+|`FORK_TIMESTAMP` | *TBD* | Mainnet |
+
+### Configuration
+
+| Name | Value | Comment |
+| - | - | - |
+| `EXIT_PRECOMPILE_ADDRESS` | *TBD* | Where to call and store relevant details about exit mechanism |
+| `EXCESS_EXITS_STORAGE_SLOT` | 0 | |
+| `EXIT_COUNT_STORAGE_SLOT` | 1 | |
+| `EXIT_MESSAGE_QUEUE_HEAD_STORAGE_SLOT` | 2 | Pointer to head of the exit message queue |
+| `EXIT_MESSAGE_QUEUE_TAIL_STORAGE_SLOT` | 3 | Pointer to the tail of the exit message queue|
+| `EXIT_MESSAGE_QUEUE_STORAGE_OFFSET` | 4 | The start memory slot of the in-state exit message queue|
+| `MAX_EXITS_PER_BLOCK` | 16 | Maximum number of exits that can be dequeued into a block |
+| `TARGET_EXITS_PER_BLOCK` | 2 | |
+| `MIN_EXIT_FEE` | 1 | |
+| `EXIT_FEE_UPDATE_FRACTION` | 17 | | 
+| `EXCESS_RETURN_GAS_STIPEND` | 2300 | |
+
+### Execution layer
+
+#### Definitions
+
+* **`FORK_BLOCK`** -- the first block in a blockchain with the `timestamp` greater or equal to `FORK_TIMESTAMP`.
+
+#### Exit operation
+
+The new exit operation consists of the following fields:
+
+1. `source_address: Bytes20`
+2. `validator_pubkey: Bytes48`
+
+RLP encoding of an exit **MUST** be computed as the following:
+
+```python
+rlp_encoded_exit = RLP([
+    RLP(source_address),
+    RLP(validator_pubkey),
+])
+```
+
+#### Exit precompile
```

Comment:
In programming, "Exit" typically conveys a system shutdown. How about renaming "Exit" to "Validator Exit"?